### PR TITLE
MessagePack deserialization: improve error messages

### PR DESCRIPTION
**Pull request by mgovers**

### CODE REVIEW

**Review by figueroa1395 — APPROVED** [review]

File: power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp [hunk 1]
```diff
@@ -193,7 +193,7 @@ struct MapArrayVisitor : DefaultErrorVisitor<MapArrayVisitor<map_array>> {
     static constexpr bool enable_array =
         std::same_as<map_array, visit_array_t> || std::same_as<map_array, visit_map_array_t>;
     static constexpr std::string_view static_err_msg =
-        enable_map ? (enable_array ? "Expect a map or array." : "Expect a map.") : "Expect an array.";
+        enable_map ? (enable_array ? "Expected a map or array." : "Expected a map.") : "Expected an array.";
```
Comment (Jerry-Jinfeng-Guo) [inline power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp:196]: Now sure this is good English. You could either do `A map or an array expected` or `Expecting a map or an array`.

**Review by Jerry-Jinfeng-Guo — CHANGES_REQUESTED** [review]
Please update the error messages: 
either "A/An {item 1}  or a/an {item 2} expected "
or 
"Expecting  a/an {item 1}  or  a/an {item 2}"

**Comment by TonyXiang8787** [comment]
What are we fixing apart from the error messages?

**Comment by mgovers** [comment]
> What are we fixing apart from the error messages?

read as (fix + improve) error messages. will update

File: power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp (hunk 1 above)
Comment (mgovers) [reply power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp:196]:
```suggestion
        enable_map ? (enable_array ? "Expected a map or an array." : "Expected a map.") : "Expected an array.";
```

**Comment by Jerry-Jinfeng-Guo** [comment]
> kept the `Expected ...` pattern. It doesn't look that weird if you read the full error image

That's exactly what I couldn't agree...

**Comment by TonyXiang8787** [comment]
I still do not get the purpose of this PR. The original error message is already very clear. The changes in this PR are only about the tense.

File: power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp [hunk 2]
```diff
@@ -248,7 +248,7 @@ struct BoolVisitor : DefaultErrorVisitor<BoolVisitor> {
 template <class T> struct ValueVisitor;
 
 template <std::integral T> struct ValueVisitor<T> : DefaultErrorVisitor<ValueVisitor<T>> {
-    static constexpr std::string_view static_err_msg = "Expect an interger.";
+    static constexpr std::string_view static_err_msg = "Expected an integer.";
```
Comment (mgovers) [inline power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp:251]: This contained a typo `interger`, fixed to `integer`. Irena encountered that

**Comment by mgovers** [comment]
> I still do not get the purpose of this PR. The original error message is already very clear. The changes in this PR are only about the tense.

In addition, while present tense was clear, it felt more like imperative tense (which it clearly isn't).

In this case, while we haven't dug into debugging yet, the JSON looked correct at first sight
* so the error message became confusing: "expect an integer" > "it tells me to expect something?"
* Past tense would have been more clear: "Expected an integer" > ah ok something went wrog. Maybe internally?

File: power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp (hunk 2 above)
Comment (TonyXiang8787) [reply power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp:251]: This change I agree. But others not. 

Change from "expect" to "expected" is a bit confusing. I propose either keep the original, or changed them to "integer expected", as @Jerry-Jinfeng-Guo suggests.

**Review by TonyXiang8787 — APPROVED** [review]

**Review by Jerry-Jinfeng-Guo — APPROVED** [review]

**Pull request merged**